File: BridgeX/app/src/main/java/cz/lriedel/bridgex/AuthenticationClient.java
package cz.lriedel.bridgex;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.POST;
import retrofit2.http.Url;

public interface AuthenticationClient {

    @POST
    Call<AccessToken> createAccessToken(@Url String url, @Body AccessTokenRequest accessTokenRequest);
}

File: BridgeX/app/src/main/java/cz/lriedel/bridgex/authentication/AuthenticationClient.kt
package cz.lriedel.bridgex.authentication

import retrofit2.http.Body
import retrofit2.http.POST
import retrofit2.http.Url

interface AuthenticationClient {
    @POST
    suspend fun createAccessToken(
        @Url url: String,
        @Body accessTokenRequest: AccessTokenRequest
    ): AccessToken
}
